homeassistant/components/shopping_list/__init__.py

```diff
@@ -66,7 +66,7 @@ def add_item_service(call):
         """Add an item with `name`."""
         data = hass.data[DOMAIN]
         name = call.data.get(ATTR_NAME)
-        if name is not None:
+        if name is not None and name != '':
```

Comment:
This should be enough:
```py
if name:
```

Reply:
We should probably catch this in the service schema already, instead.

Reply:
All right. I'll figure out how to do that.